app/models/case_log.rb

```diff
@@ -461,9 +468,19 @@ def reset_invalidated_dependent_fields!
   end
 
   def dynamically_not_required
-    previous_la_known_field = postcode_known? ? %w[previous_la_known] : []
-    tshortfall_field = tshortfall_unknown? ? %w[tshortfall] : []
-    previous_la_known_field + tshortfall_field
+    not_required = []
+    not_required << "previous_la_known" if postcode_known?
+    not_required << "tshortfall" if tshortfall_unknown?
+    not_required << "tenancylength" if tenancylength_optional?
+
+    not_required
+  end
+
+  def tenancylength_optional?
+    return false unless collection_start_year
+    return true if collection_start_year < 2022
+
+    collection_start_year >= 2022 && ![4, 6].include?(tenancy)
```

Comment:
Could you not call `!is_secure_tenancy?` here?

Reply:
Secure tenancy is 6 and 7. This is fixed term, I'll make a helper.